packages/ui/src/components/pages/SetupWallet/NewWallet/BackupWallet.tsx

```diff
@@ -0,0 +1,129 @@
+
+interface BackupProps extends Props {
+  method: WalletRecoveryMethod;
+}
+
+function Backup({ method }: BackupProps): JSX.Element {
+  const { t } = useTranslation();
+  const { password, secretPhrase } = useSelector((state: RootState) => state.setupWallet);
+
+  // Preventing re-create the wallet backup when the component is re-rendered
+  const walletBackup = useMemo(() => {
+    const encryptedMnemonic = CryptoJS.AES.encrypt(secretPhrase!, password!).toString();
+
+    return {
+      encryptedMnemonic,
+    } as WalletBackup;
+  }, [secretPhrase, password]);
+
+  switch (method) {
+    case WalletRecoveryMethod.SecretRecoveryPhrase:
+      return <BackupSecretRecoveryPhrase secretPhrase={secretPhrase!} title='Secret Recovery Phrase' />;
+    case WalletRecoveryMethod.QrCode:
+      return (
+        <QrCode
+          value={walletBackup}
+          object={TransferableObject.Wallet}
+          title='Downloads QR Code'
+          topInstruction={
+            <p>
+              {t<string>(
+                'Downloads the below QR code image and upload it to the cloud (iCloud, Google Drive, ...), you can restore your wallet by scanning or uploading this QR Code to Coong Wallet later at any time with your wallet password.',
+              )}
+            </p>
+          }
+          bottomInstruction={<></>} // Use to hide default bottom instruction
+        />
+      );
+    case WalletRecoveryMethod.JsonFile:
+      return (
+        <JsonFile
+          value={walletBackup}
+          object={TransferableObject.Wallet}
+          title='Downloads JSON File'
+          topInstruction={
+            <p>
+              {t<string>(
+                'Downloads the below JSON file and upload it to the cloud (iCloud, Google Drive, ...), you can restore your wallet by uploading this JSON file to Coong Wallet later at any time with your wallet password.',
+              )}
+            </p>
+          }
+          bottomInstruction={<></>} // Use to hide default bottom instruction
+        />
+      );
+    default:
+      return <></>;
+  }
+}
+
+interface BackupWalletProps extends Props {
+  method: WalletRecoveryMethod;
+  resetMethod: () => void;
+}
+
+export default function BackupWallet({ method, resetMethod }: BackupWalletProps): JSX.Element {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const { password, secretPhrase } = useSelector((state: RootState) => state.setupWallet);
+  const { setup, loading } = useSetupWallet({ secretPhrase, password });
+  const [checked, setChecked] = useState<boolean>(false);
+
+  useEffectOnce(() => {
+    if (secretPhrase) return;
+    dispatch(setupWalletActions.setSecretPhrase(generateMnemonic(12)));
+  });
+
+  const handleCheckbox = (event: ChangeEvent<HTMLInputElement>) => {
+    setChecked(event.target.checked);
+  };
+
+  const doSetupWallet = (e: FormEvent) => {
+    e.preventDefault();
+
+    setup();
+  };
+
+  return (
+    <>
+      <Backup method={method} />
```

Comment:
Probably rename this to `BackupInstruction` for clarity!
```suggestion
      <BackupInstruction method={method} />
```